src/pages/home/home.ts

```diff
@@ -278,42 +330,71 @@ export class HomePage {
-      //return resolve({coords: { latitude: 25.992046, longitude: -80.383645 }})
     });
   }
 
-  getCentralPosition(currentPosition: Coordinates) {
-    // const locations = [currentPosition, ...RANDOM_GEOCOORDINATES];
+  adjustCentralLocation(currentPosition) {
+    //Find old central location marker.
+    let centralLocationMarker = this.droppedMarkers.find(
+      x => x.title === 'Central Location'
+    );
 
-    return new Promise(resolve => {
-      // const centralPosition = geolib.getCenterOfBounds(locations);
+    //Remove old central location marker.
+    if (centralLocationMarker) {
+      centralLocationMarker.marker.remove();
+      _.pull(this.droppedMarkers, centralLocationMarker);
+    }
 
-      // TODO: should be dropping the pin on the central location
-      // but for now it will just be the current position
+    //Share central position observable that calculates central position among users.
+    let centralPosObservable = this.getCentralPosition(currentPosition).share();
 
-      this.onWaterService
-        .checkForWater(currentPosition.latitude, currentPosition.longitude)
-        .subscribe(res => {
-          this.isOnWater = res.json().water;
-          if (this.isOnWater === true) {
-            alert(
-              'It looks like the central location is on water! You have the chance to move the pin and put it on land.'
-            );
-          }
-          this.dropMarker(
-            'Central Location',
-            'purple',
-            currentPosition.latitude,
-            currentPosition.longitude,
-            true
-          );
-        });
+    centralPosObservable.subscribe(centralPosition =>
+      this.getWorkfromLocations(centralPosition)
+    );
+    centralPosObservable.subscribe(centralPosition =>
+      this.checkIfPositionOnWater(centralPosition)
+    );
+  }
 
-      return resolve(currentPosition);
+  getCentralPosition(currentPosition) {
+    //Array of user locations.
+    const userLocations: PositionAsDecimal[] = this.groupSocketService.userInfos.map(
+      x => {
+        return { latitude: x.latitude, longitude: x.longitude };
+      }
+    );
+    const locations = [currentPosition, ...userLocations];
+    console.log(`${JSON.stringify(locations)}`);
+    return new Observable(observer => {
+      const centralPosition = geolib.getCenterOfBounds(locations);
+      this.dropMarker(
+        'Central Location',
+        this.generateMarkerUID(),
+        'purple',
+        centralPosition.latitude,
+        centralPosition.longitude,
+        this.isOnWater
+      );
+      alert(`CENTRAL POS IS: ${JSON.stringify(centralPosition)}`);
+      observer.next(centralPosition);
     });
   }
 
+  checkIfPositionOnWater(centralPosition) {
+    this.onWaterService
+      .checkForWater(centralPosition.latitude, centralPosition.longitude)
+      .subscribe(res => {
+        this.isOnWater = res.json().water;
+        if (this.isOnWater === true) {
+        }
```

Comment:
We are not doing anything in this if, looks like the alert that we had here was removed my accident.

Reply:
The alert is
```
alert('It looks like the central location is on water! You have the chance to move the pin and put it on land.');
```

Reply:
Ah, so that's why it wasn't popping up when the location was in the lake!